packages/api/app/Controllers/Http/PermissionController.js

```diff
@@ -13,9 +13,7 @@ class PermissionController {
 	 * GET permissions
 	 */
 	async index({ request }) {
-		return Permission.query()
-			.withParams(request.params)
-			.fetch();
+		return Permission.query().withParams(request);
```

Comment:
Não acho que devemos mover o `fetch()` para o `withParams` é interessante manter aqui mesmo pois nos permite alterar a query aqui para casos bem específicos.

Reply:
Preciso do `fetch` dentro do `withParams` para fazer a paginação. Usei o mesmo principio do paginate, quando chamar o  `withParams` o query deve está completo.

https://adonisjs.com/docs/4.1/lucid#_pagination

O `firstOrFail` pode ficar fora do `withParams`,  chamei `firstOrFail` dentro do `withParams` por que observei o mesmo comportamento em todas as rotas **show** e já que estou enviando o request para dentro do Trait tentei tratar o parâmetro **ID** lá.

Reply:
Entendi. Nesse caso se for pra mantar o `fetch` é melhor manter ambos mesmo.